ci/conform-nvim.nix

```
@@ -1,13 +1,11 @@
 {
   vimPlugins,
   lib,
-  writeText,
 }:
 lib.pipe "${vimPlugins.conform-nvim}/lua/conform/formatters" [
   builtins.readDir
   builtins.attrNames
   (builtins.filter (lib.hasSuffix ".lua"))
   (map (lib.removeSuffix ".lua"))
-  builtins.toJSON
-  (writeText "conform-formatters")
+  (lib.writers.writeJSON "conform-formatters.json")
```

`writers` is part of `pkgs`, not `lib`. We should get it from the callPackage args above.

```suggestion
  (writers.writeJSON "conform-formatters.json")
```

---

> For some reason this writer creates slightly differently formatted JSON than writeText + toJSON

Makes sense. It is using `jq` to normalise the JSON: https://github.com/NixOS/nixpkgs/blob/852841c072353c343ba2fdc70d02469da3be886e/pkgs/pkgs-lib/formats.nix#L151-L164

> prettier is input dependant

That is cursed... Are you saying that `prettier` and `treefmt` are now producing different output?

I'm fine with the formatting being different from before, so long as `treefmt` remains idempotent.

---

> Are you saying that prettier and treefmt are now producing different output?

No! `writers.writeJSON` and `writeText (toJSON)` produce differently formatted JSON. So the diff on the generation changed just now. Prettier and treefmt produce the same output. I was just mentioning that despite both being formatted the formatting changed, because prettier output depends on the initial formatting.